homeassistant/components/plugwise/climate.py

```diff
@@ -87,24 +104,41 @@ def hvac_mode(self) -> HVACMode:
             return HVACMode.HEAT
         return HVACMode(mode)
 
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the current hvac modes."""
+        hvac_modes = [HVACMode.HEAT]
+        if self.coordinator.data.gateway["cooling_present"]:
+            if self.hc_data.get("elga_cooling_enabled", False):
+                hvac_modes.append(HVACMode.HEAT_COOL)
+                hvac_modes.remove(HVACMode.HEAT)
+            if self.hc_data.get("lortherm_cooling_enabled", False) or self.hc_data.get(
+                "adam_cooling_enabled", False
+            ):
+                hvac_modes.append(HVACMode.COOL)
+                hvac_modes.remove(HVACMode.HEAT)
+        if self.device["available_schedules"] != ["None"]:
+            hvac_modes.append(HVACMode.AUTO)
+
+        return hvac_modes
+
     @property
     def hvac_action(self) -> HVACAction:
         """Return the current running hvac operation if supported."""
         # When control_state is present, prefer this data
-        if "control_state" in self.device:
-            if self.device.get("control_state") == "cooling":
-                return HVACAction.COOLING
-            # Support preheating state as heating, until preheating is added as a separate state
-            if self.device.get("control_state") in ["heating", "preheating"]:
-                return HVACAction.HEATING
-        else:
-            heater_central_data = self.coordinator.data.devices[
-                self.coordinator.data.gateway["heater_id"]
-            ]
-            if heater_central_data["binary_sensors"].get("heating_state"):
-                return HVACAction.HEATING
-            if heater_central_data["binary_sensors"].get("cooling_state"):
-                return HVACAction.COOLING
+        control_state: str = self.device.get("control_state", "not_found")
+        if control_state == "cooling":
+            return HVACAction.COOLING
+        # Support preheating state as heating, until preheating is added as a separate state
+        if control_state in ["heating", "preheating"]:
+            return HVACAction.HEATING
+        if control_state == "off":
+            return HVACAction.IDLE
+
+        if self.hc_data["binary_sensors"]["heating_state"]:
+            return HVACAction.HEATING
+        if self.hc_data["binary_sensors"].get("cooling_state", False):
```

Comment:
None will evaluate as falsy too...

```suggestion
        if self.hc_data["binary_sensors"].get("cooling_state"):
```